### PR TITLE
feat: Enable cost analysis add-on in managed cluster module

### PR DESCRIPTION
**Pull request by PixelRobots**
## Description

This pull request adds the ability to enable the newly GA cost analysis feature of AKS.

## Pipeline Reference

| Pipeline |
| -------- |
|     [avm.res.container-service.managed-cluster](https://github.com/PixelRobots/bicep-registry-modules/actions/workflows/avm.res.container-service.managed-cluster.yml)     |

## Type of Change

- [ ] Update to CI Environment or utlities (Non-module effecting changes)
- [x] Azure Verified Module updates:
  - [ ] Bugfix containing backwards compatible bug fixes, and I have NOT bumped the MAJOR or MINOR version in `version.json`:
    - [ ] Someone has opened a bug report issue, and I have included "Closes #{bug_report_issue_number}" in the PR description.
    - [ ] The bug was found by the module author, and no one has opened an issue to report it yet.
  - [ ] Feature update backwards compatible feature updates, and I have bumped the MINOR version in `version.json`.
  - [ ] Breaking changes and I have bumped the MAJOR version in `version.json`.
  - [ ] Update to documentation

## Checklist

- [x] I'm sure there are no other open Pull Requests for the same update/change
- [x] I have run `Set-AVMModule` locally to generate the supporting module files.
- [ ] My corresponding pipelines / checks run clean and green without any errors or warnings

### CODE REVIEW

File: avm/res/container-service/managed-cluster/main.bicep [hunk 1]
```diff
@@ -716,7 +719,7 @@ resource managedCluster 'Microsoft.ContainerService/managedClusters@2023-07-02-p
         enabled: enableStorageProfileBlobCSIDriver
       }
       diskCSIDriver: {
-        enabled: enableStorageProfileDiskCSIDriver
+        enabled: costAnalysisEnabled == true ? true : enableStorageProfileDiskCSIDriver
```
Comment (ilhaan) [inline avm/res/container-service/managed-cluster/main.bicep:722]: @PixelRobots Why is `costAnalysisEnabled` being used for `diskCSIDriver`? This is supposed to go under `metricsProfile` as per docs: https://learn.microsoft.com/en-us/azure/templates/microsoft.containerservice/managedclusters?pivots=deployment-language-bicep#managedclustermetricsprofile

**Review by ilhaan — CHANGES_REQUESTED** [review]

File: avm/res/container-service/managed-cluster/main.bicep (hunk 1 above)
Comment (PixelRobots) [reply avm/res/container-service/managed-cluster/main.bicep]: I have added it under metrics profiule. i forgot to copy and paste that over from my master. (i have a lot of changes I can add here if needed, like keda and more). 

I have also added the skuTier check on the storage as thios feature needs the disk driver to be enabled, but also it does not work on the free tier of AKS.

Comment (PixelRobots) [reply avm/res/container-service/managed-cluster/main.bicep]: Thanks. I will start dropping them in.

I have been working on a user defined type and module for Aks maintenance windows configuration but not sure it is 100% the best but will share that one too.

**Review by ilhaan — APPROVED** [review]
@PixelRobots thanks for the PR!

**Comment by ilhaan** [comment]
@PixelRobots Would it be possible for you to run our validation pipeline for these module changes and add a status badge to the PR description? You can find instructions for doing this here: https://azure.github.io/Azure-Verified-Modules/contributing/bicep/bicep-contribution-flow/#1-setup-your-azure-test-environment. 

No worries if you can't do this. I can run them for you but will probably only get to this on Thursday or so. 

CC: @matebarabas @JPEasier

**Comment by PixelRobots** [comment]
Added status badge now.

**Comment by PixelRobots** [comment]
It was showing as pass before. not too sure why its failed now. may have been due to me cancelling a job in my repo on main.

**Comment by ilhaan** [comment]
@PixelRobots I think you should be able to create a status badge for the specific run of the main branch that had this cost analysis stuff passing.

**Comment by PixelRobots** [comment]
Unfortunately not, or I don't know how. I have kicked off another run on the main branch so in about an hour it should be green.

**Comment by AlexanderSehr** [comment]
> Unfortunately not, or I don't know how. I have kicked off another run on the main branch so in about an hour it should be green.

Hey @PixelRobots just a quick note to avoid potential confusion with failing workflows. As your contribution was opened from your main branch, the workflow will also run actions that would not run for any other branch. In particular one - the publishing. 

Naturally, you cannot publish from your fork into the Public Registry so that step will fail. It doesn't say anything about your contribution, as at that point, all previous tests already passed. Just doesn't look that pretty.

There's soon going to be a contribution that will skip this step, but for the time being, either ignore the red pipeline if only the publishing failed, or create contributions in branches 💪 

cc: @eriqua, @ilhaan

**Comment by AlexanderSehr** [comment]
One last note @PixelRobots,
as I don't want you to sigh in frustration. I just took a look at your workflow run and saw an error massage like 
> dep***diasacsmaz01 is not a valid storage account name. Storage account name must be between 3 and 24 characters in length and use numbers and lower-case letters only.

That bad boy usually implies that either the `namePrefix` secret was not correctly configured in the CI environment - or - that it is too long. I think the recommendation is 3-5 characters, and the only reason for this limit are resources like the storage acount that have a very short name limitation. Please doublecheck if the value is not too long.

**Comment by PixelRobots** [comment]
Thanks will make it smaller. 

I figured out the mainland branch thing. My other pr which is built of my main which includes this change passed and is all green. 

So I know it works. I just want the green lol.

**Comment by PixelRobots** [comment]
And we have Green!

**Comment by AlexanderSehr** [comment]
Hey @PixelRobots is there a reason you did not run the static tests?

**Comment by ilhaan** [comment]
Thank you @PixelRobots!

**Comment by PixelRobots** [comment]
> Hey @PixelRobots is there a reason you did not run the static tests? 

It had issues creating a file on the runner even with read and write permissions.

**Comment by ilhaan** [comment]
@AlexanderSehr I missed that static tests did not run. I noticed that `Set-AVMModule` was not run to generate a new main.json, so I created a PR for that here: https://github.com/Azure/bicep-registry-modules/pull/1695. 

CC: @PixelRobots

**Pull request merged**